### DISCUSSION TITLE
Javascript/Typescript LSP in templ files?

### DISCUSSION THREAD

**Discussion by Giftzwerg02** [a-h/templ#917] [Q&A]
Is it possible to activate tsserver in script blocks inside templ files?
i.e.

![image](https://github.com/user-attachments/assets/b9ff907b-614a-4072-ae08-4d3819ff0789)

I would like to have auto-completion here...

I didn't find any related issue or discussion to this yet, I am curious if anyone already has this figured out. I am using neovim, but I guess any editor that support LSP should be equally suited for this.

**Comment by Smithx10** [comment]
I wouldn't mind attempting to implement this,  I'd like to learn more about lsp and this would be a useful way to do it. @a-h What would this look like?   

I've successfully parsed the templateFile and found all the script_tag elements... @Giftzwerg02  as long as I don't get burned out I think I will push a branch.

Not sure if anyone has done this work already before.

**Comment by Smithx10** [comment]
I was able to plumb in some of this with typescript-language-server for my own use case.  
After I've fully comprehended how this all works,  I should be able to push a branch.

![image](https://github.com/user-attachments/assets/8b73f79e-d7e2-4ef2-a4ef-cb53813058ea)
![image](https://github.com/user-attachments/assets/cc0dd6a7-d01f-44c9-a5ea-8e3e32a55b47)
![image](https://github.com/user-attachments/assets/7e26c95e-1a86-4cfe-911b-d27892d38c11)
![image](https://github.com/user-attachments/assets/6df1c301-b8e5-463b-a8fd-a320b4b05ee2)

**Reply by Giftzwerg02** [reply]
Looks amazing! I currently have a bit more time on my hands, so if you need help with something just hmu :+1:
